File: q5.js
const readline = require('readline'); // Livraria para digitar no console.
// Lembrando que "readline" é uma livraria inata do NodeJS

// Cria uma interface para leitura de dados do console
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function inverteString(s) {
    /*
    Função que inverte os caracteres de uma string.
    */
    let caracteres = s.split(''); // Converte a string em um array de caracteres

    let inicio = 0; // Índice inicial para inverter o array
    let fim = caracteres.length - 1; // Índice final para inverter o array

    while (inicio < fim) {
        // Troca o caractere no índice inicio com o caractere no índice fim
        [caracteres[inicio], caracteres[fim]] = [caracteres[fim], caracteres[inicio]];
        // Atualiza os índices
        inicio++;
        fim--;
    }

    let sInvertida = caracteres.join(''); // Converte o array de caracteres de volta para uma string

    return sInvertida;
}

// Solicita que o usuário digite a palavra a ser invertida
rl.question("Digite a palavra a ser invertida: ", function(stringOriginal) {
    // Chama a função inverteString para inverter a string digitada
    let stringInvertida = inverteString(stringOriginal);

    console.log("String original: " + stringOriginal);
    console.log("String invertida: " + stringInvertida);

    rl.close(); // Fecha a interface de leitura de dados do console
});

File: q5.py
def inverte_string(s):
    """
    Função que inverte os caracteres de uma string.
    """
    # Converte a string em uma lista de caracteres
    caracteres = list(s)
    
    # Índices inicial e final para inverter a lista
    inicio = 0
    fim = len(caracteres) - 1
    
    # Inverte os caracteres da lista usando o índice
    while inicio < fim:
        # Troca o caractere no índice inicio com o caractere no índice fim
        caracteres[inicio], caracteres[fim] = caracteres[fim], caracteres[inicio]
        # Atualiza os índices
        inicio += 1
        fim -= 1
    
    # Converte a lista de caracteres de volta para uma string
    s_invertida = ''.join(caracteres)
    
    return s_invertida

# Solicita que o usuário digite a palavra a ser invertida
string_original = input("Digite a palavra a ser invertida: ")

# Chama a função inverte_string para inverter a string digitada
string_invertida = inverte_string(string_original)

print("String original: ", string_original)
print("String invertida: ", string_invertida)
